### ISSUE TITLE
Covariance and error checking

### ISSUE THREAD

**Issue by drchrislewis** [ros-industrial/industrial_calibration#42]
The calibration can run to completion, but get poor results. There are three possible reasons
1. The optimization is caught in a local minimum.
2. The optimization is ill conditioned or not fully constrained
3. The optimization found an alternate symmetric solution.

The first case usually has large residual error. We should check and flag this situation
The second case is detectable through covariance analyis the Jacobian will not be full rank.
For small problems, this can be checked through Cere's covariance 
The third case is difficult to detect, except that the results are absurd. For example, mathematically, the same image of a target is obtained by placing the camera on the opposite side of the target and facing it away from the object. Similar difficulties occur when the target is symetric. The same image of the target is possible from as many as 8 different vantage points, both in front of and behind the target. 
For extrinsic calibration, having a camera either behind a target, or facing away from it could be detected and flagged. 

In short we need error checking and flagging.

**Comment by drchrislewis** [comment]
latest pull should cover this for the most part

**Comment by drchrislewis** [comment]
This important feature exist, but should be documented with examples so that even the casual user can investigate and be confident in their results.
